DesignSafe-CI/DS-User-Guide
# feat: use taccwma/tacc-docs docker image as base ⚠️

## Overview

Use https://github.com/TACC/TACC-Docs as base for Docker-container–based development.

[TACC]: https://github.com/TACC/TACC-Docs

> [!IMPORTANT]
> This changes `/docs` to `/code`, like https://github.com/TACC/TACC-Docs/commit/d407a0d which required https://github.com/TACC/Core-Portal-Deployments/commit/b028906. So, before merge, definitely mimic https://github.com/TACC/Core-Portal-Deployments/commit/b028906 but for [DesignSafe Doc deployment](https://github.com/DesignSafe-CI/ansible/pull/15/files).

> [!WARNING]
> Main branch **uninstalls** a plugin that TACC Docs still uses, and installs a replacement plugin that TACC Docs does **not** use. I dunno how to resolve this yet. — @wesleyboar 2024-02-22

> [!CAUTION]
> MkDocs `build`/`serve` fails because [`macros` plugin](https://mkdocs-macros-plugin.readthedocs.io/) errors when it encounters Python code in a document. This repo has this often, while TACC Docs does not. (If I add Python code to a TACC-Docs document, it fails with the same error.) I think the `mkdocs-macros-plugin` only exists for `{% include … %}` which can be replaced by [`pymdownx.snippets`](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/). But that brings up a problem that `main` branch also suffers from: the `pymdownx.snippets` requires rewriting image paths, which requires suffering build warnings.

## Related

- [DES-2469](https://tacc-main.atlassian.net/browse/DES-2469)
- requires https://github.com/TACC/TACC-Docs/pull/15

## Changes

- **added** rule to ignore TACC-Docs files in Git
- **removed** most of Dockerfile content
- **changed** Dockerfile to:
    1. load `taccwma/tacc-docs`
    2. install `yq`
    3. archive and replace TACC documents
- **added** Makefile and instructions to use it
- **added** code to integrate TACC docs _**only** for Docker users_
- **removed** Poetry dependencies (which TACC provides)
- **added** favicon, logo, and CMS link

## Testing

1. `make start`
2. `make build`
3. Verify site loads.
4. Verify site has style of https://docs.tacc.utexas.edu.
5. Verify site has content of https://designsafeci-dev.tacc.utexas.edu/user-guide/.
6. Verify search still works.

## UI

![designsafe using tacc docs styles](https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/2a209583-c51c-44a4-903f-70881eb06f9b)
